src/libs/DistanceRequestUtils.js

```diff
@@ -0,0 +1,93 @@
+ * @param {'mi' | 'ki'} [policy.customUnits.attributes.unit] - unit of measurement for the distance
+ *
+ * @returns {Object|null} An object containing the rate and unit for the default mileage or null if not found.
+ * @returns {Number} .rate - The default rate for the mileage.
+ * @returns {String} .unit - The unit of measurement for the distance.
+ */
+const getDefaultMileageRate = (policy) => {
+    if (!policy || !policy.customUnits) {
+        return null;
+    }
+
+    const distanceUnit = _.find(_.values(policy.customUnits), (unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    if (!distanceUnit) {
+        return null;
+    }
+
+    const distanceRate = _.find(_.values(distanceUnit.rates), (rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
+    if (!distanceRate) {
+        return null;
+    }
+
+    return {
+        rate: distanceRate.rate,
+        unit: distanceUnit.attributes.unit,
+    };
+};
+
+/**
+ * Converts a given distance in meters to the specified unit (kilometers or miles).
+ *
+ * @param {Number} distanceInMeters - The distance in meters to be converted.
+ * @param {'mi' | 'ki'} unit - The desired unit of conversion, either 'km' for kilometers or 'mi' for miles.
+ *
+ * @returns {Number} The converted distance in the specified unit.
+ */
+function convertDistanceUnit(distanceInMeters, unit) {
+    const METERS_TO_KM = 0.001; // 1 kilometer is 1000 meters
+    const METERS_TO_MILES = 0.000621371; // There are approximately 0.000621371 miles in a meter
+
+    switch (unit) {
+        case CONST.CUSTOM_UNITS.DISTANCE_UNIT_KILOMETERS:
+            return distanceInMeters * METERS_TO_KM;
+        case CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES:
+            return distanceInMeters * METERS_TO_MILES;
+        default:
+            throw new Error('Unsupported unit. Supported units are "mi" or "km".');
+    }
+}
+
+/**
+ *
+ * @param {Number} distanceInMeters Distance traveled
+ * @param {'mi' | 'ki'} unit Unit that should be used to display the distance
+ * @param {Number} rate Expensable amount allowed per unit
+ * @param {Function} translate Translate function
+ * @returns {String} A string that describes the distance travled and the rate used for expense calculation
+ */
+const getDistanceString = (distanceInMeters, unit, rate, translate) => {
+    const convertedDistance = convertDistanceUnit(distanceInMeters, unit);
+
+    const distanceUnit = unit === CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES ? translate('common.miles') : translate('common.kilometers');
+    const singularDistanceUnit = unit === CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES ? translate('common.mile') : translate('common.kilometer');
+    const roundedDistance = convertedDistance.toFixed(2);
+    const unitString = roundedDistance === 1 ? singularDistanceUnit : distanceUnit;
+    const ratePerUnit = rate * 0.01;
+
+    return `${roundedDistance} ${unitString} @ $${ratePerUnit} / ${singularDistanceUnit}`;
```

Comment:
BUG: The money request amount on the confirmation page is always shown in dollars.
1. Open Settings > Workspaces > select your test workspace > settings > change the default currency to non-USD
2. Start to create a distance request and stop on the confirmation page

Expected result: the amount is displayed in the workspace currency
Actual result: it's displayed as USD